src/cache/tlru-cache.spec.ts

```diff
@@ -22,22 +22,65 @@ describe('TLRU Cache', () => {
     expect(cache.get('a')).toBeUndefined();
   });
 
-  it('should evict all expired entries', () => {
+  it('should not evict cache before expiration', () => {
+    jest.useFakeTimers();
+
+    cache.set('a', 'apple');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs - 1);
+    expect(cache.get('a')).toBe('apple');
+  });
+
+  it('should evict all expired entries on .entries() call', () => {
     jest.useFakeTimers();
 
     cache.set('a', 'avocado');
     jest.advanceTimersByTime(expectedCacheTimeoutMs);
     cache.set('b', 'banana');
     jest.advanceTimersByTime(expectedCacheTimeoutMs);
 
-    expect(cache.get('b')).toBeUndefined();
-    expect(cache.get('a')).toBeUndefined();
+    const cacheEntries = [];
+
+    for (const entry of cache.entries()) {
+      cacheEntries.push(entry);
+    }
+
+    expect(cacheEntries.length).toBe(0);
+  });
+
+  it('should evict all expired entries on .keys() call', () => {
+    jest.useFakeTimers();
+
+    cache.set('a', 'avocado');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs);
+    cache.set('b', 'banana');
+    jest.advanceTimersByTime(expectedCacheTimeoutMs);
+
+    const cacheKeys = [];
+
+    for (const key of cache.keys()) {
+      cacheKeys.push(key);
+    }
+
+    expect(cacheKeys.length).toBe(0);
```

Comment:
probably good to leave one non-expired key in here to make sure it's still there